test/run_test.py

```diff
@@ -628,6 +630,9 @@ def run_doctests(test_module, test_directory, options):
     Assumes the incoming test module is called doctest, and simply executes the
     xdoctest runner on the torch library itself.
     """
+    # Maybe there's a better place to do this skip?
+    if is_slow_gradcheck_env():
```

Comment:
Maybe prevent this function from being called altogether?

Reply:
hmm now that I'm reading the test selection code, I don't think it's too much better there. Will keep here for now.

Reply:
I think we can follow the pattern of
```
    if not torch._C.has_lapack:
        selected_tests = exclude_tests(TESTS_REQUIRING_LAPACK, selected_tests,
                                       "PyTorch is built without LAPACK support.")
```
in `get_selected_tests()`, e.g. have something like `TESTS_NOT_USING_GRADCHECK`. This would let us collocate all the skips for slow gradcheck.

Reply:
I'll probably do a little more refactoring/cleaning up after we verify that these changes help stop the timeout.

Reply:
Sounds good!